lib/galaxy/datatypes/binary.py

```diff
@@ -60,7 +81,11 @@
 )
 
 if TYPE_CHECKING:
-    from galaxy.model import DatasetInstance
+    from galaxy.datatypes.util.complression_utils import FileObjType
```

Comment:
```suggestion
    from galaxy.datatypes.util.compression_utils import FileObjType
```

Reply:
Is this not going through type checks ? This should've made something fail ...

Reply:
If you think about the type checking that happens in the packages when we import say something from structured app in a package that doesn't depend on galaxy-app (lower in the dependency tree) it doesn't fail. When I first saw people adding those imports - I thought for sure they would fail but they don't - I guess type checking imports are all optional? I imagine there is a mypy flag for this - but there are places in the project where we now depend on this behavior.

Reply:
The path was incorrect too: it's `galaxy.util`, not `galaxy.datatypes.util`. Fixed.

I have no idea why it didn't fail. The docs (both mypy and python, PEP484) offer no clarification. I guess it's entirely up to mypy how to handle these imports. Apparently, you can `import gibberish from gibberish`, then do anything with `gibberish` inside an annotated function that is type-checked, and it'll pass just fine. But if you do the same for a real import (still inside the TYPE_CHECKING block), and then try using a non-existent attribute, then it'll raise an error. Same for trying `from galaxy import gibberish`. So, it seems, it ignores imports it cannot find and treats them as `Any`.